internal/model/model.go

```diff
@@ -84,8 +84,12 @@ type service interface {
 }
 
 type Model struct {
-	cfg             *config.Wrapper
-	db              *leveldb.DB
+	cfg *config.Wrapper
+	db  *leveldb.DB
+
+	cache    *db.NamespacedKV
+	cacheMut sync.Mutex
```

Comment:
Also, if not, why not handle locking inside the NamespacedKV, with proper Locks and RLocks where relevant?
I guess the argument is to acquire lock once rather than long term.